couch/tests/conftest.py

```diff
@@ -84,82 +115,83 @@ def send_replication(couch_version):
+def enable_cluster():
+    auth = (common.USER, common.PASSWORD)
     headers = {'Accept': 'text/json'}
 
-    # Generate a test database
-    requests.put("{}/kennel".format(common.URL), auth=auth, headers=headers)
-
-    # Populate the database
-    data = {
-        "language": "javascript",
-        "views": {
-            "all": {"map": "function(doc) { emit(doc._id); }"},
-            "by_data": {"map": "function(doc) { emit(doc.data, doc); }"},
-        },
-    }
-    requests.put("{}/kennel/_design/dummy".format(common.URL), json=data, auth=auth, headers=headers)
-
-    urls = [
-        "{}/_node/node1@127.0.0.1/_stats".format(common.URL),
-        "{}/_node/node2@127.0.0.1/_stats".format(common.URL),
-        "{}/_node/node3@127.0.0.1/_stats".format(common.URL),
-    ]
-
-    ready = defaultdict(bool)
-    for _ in range(120):
-        print("Waiting for stats to be generated on the nodes...")
-        try:
-            for url in urls:
-                if not ready[url]:
-                    res = requests.get(url, auth=auth, headers=headers)
-                    if res.json():
-                        ready[url] = True
-            if len(ready) and all(ready.values()):
-                break
-        except Exception:
-            pass
-        sleep(1)
-
-    if couch_version == "1":
-        return
-
-    doc_url = "{}/_replicator/_all_docs".format(common.URL)
-    for _ in range(120):
-        try:
-            res = requests.get(doc_url, auth=auth, headers=headers)
-            data = res.json()
-            if data.get('rows'):
-                break
-        except Exception:
-            pass
-        sleep(1)
+    requests_data = []
+    for node in ["couchdb-1.example.com", "couchdb-2.example.com"]:
+        requests_data.append(
+            {
+                "action": "enable_cluster",
+                "username": common.USER,
+                "password": common.PASSWORD,
+                "bind_address": "0.0.0.0",
+                "port": 5984,
+                "node_count": 3,
+                "remote_node": node,
+                "remote_current_user": common.USER,
+                "remote_current_password": common.PASSWORD,
+            }
+        )
+        requests_data.append(
+            {
+                "action": "add_node",
+                "username": common.USER,
+                "password": common.PASSWORD,
+                "host": node,
+                "port": 5984,
+                "singlenode": False,
+            }
+        )
+
+    for data in requests_data:
+        resp = requests.post("{}/_cluster_setup".format(common.URL), json=data, auth=auth, headers=headers)
+        resp_data = resp.json()
+        print('[INFO] cluster setup resp', resp_data)
+
+    resp = requests.get("{}/_membership".format(common.URL), auth=auth, headers=headers)
+    membership = resp.json()
+    print("[INFO] membership response:", membership)
+    assert len(membership['cluster_nodes']) == 3
```

Comment:
This will make that whole funciton be retried on failure. Is that a problem to submit multiple request to `_cluster_setup`?

Reply:
Yes, that's the intent in case there are some issues with the cluster setup calls. Retrying should be ok (at least for testing).

We are only interested by the end result, which is having those 3 `cluster_nodes`.